### PR TITLE
✨ Add DM related actions and data

### PR DESCRIPTION
**Pull request by foysalit**
In preparation for the DM feature, this PR gives ozone capabilities to 

1. Enable/Disable DMs for a user through tags.
2. View messages exchanged between 2 users within a certain window around a reported message.
3. View stats around DM usage by a user.

### CODE REVIEW

File: components/mod-event/EventItem.tsx [hunk 1]
```diff
@@ -5,8 +5,10 @@ import { ReasonBadge } from '@/reports/ReasonBadge'
 import {
   ToolsOzoneModerationDefs,
   ComAtprotoModerationDefs,
+  ChatBskyConvoDefs,
```
Comment (devinivy) [inline components/mod-event/EventItem.tsx:8]: The API package is published with access to these lexicons 👍

File: components/mod-event/DetailsPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+              </div>
+            </Popover.Panel>
+          </Transition>
+        </>
+      )}
+    </Popover>
+  )
+}
+
+const ModEventDetails = ({ modEventType }: { modEventType: string }) => {
```
Comment (devinivy) [inline components/mod-event/DetailsPopover.tsx:48]: Good content in here!

File: components/mod-event/constants.ts [hunk 3]
```diff
@@ -14,6 +14,9 @@ export const MOD_EVENTS = {
   EMAIL: 'tools.ozone.moderation.defs#modEventEmail',
   TAG: 'tools.ozone.moderation.defs#modEventTag',
   DIVERT: 'tools.ozone.moderation.defs#modEventDivert',
+  APPEAL: 'appeal',
```
Comment (devinivy) [inline components/mod-event/constants.ts:17]: Just a vibe thing, but I expected to encounter more logic associated with this new constant.  So just double checking it's all handled 👍

File: components/dms/MessageActorMeta.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    cacheTime: 10 * 60 * 1000,
+    staleTime: 10 * 60 * 1000,
+    retry: 0,
+    enabled,
+    queryKey: ['messageActorMeta', { did }],
+    queryFn: async () => {
+      // TODO: Use this instead of the mock data
+      // const { data } = await client.api.chat.bsky.moderation.getActorMetadata(
+      // { actor: did },
+      // { headers: client.proxyHeaders() },
```
Comment (devinivy) [inline components/dms/MessageActorMeta.tsx:51]: We'll want the proxy headers to reference the backing chat service.

**Review by devinivy — COMMENTED** [review]
I know there are still a few things to button up, but this is looking good 👍

File: components/dms/MessageActorMeta.tsx (hunk 4 above)
Comment (devinivy) [reply components/dms/MessageActorMeta.tsx:51]: Oops, forget I said that!  The request will go through ozone, which will perform service auth with the chat service.  So this looks correct 👍

File: components/mod-event/constants.ts (hunk 3 above)
Comment (foysalit) [reply components/mod-event/constants.ts:17]: yep yep, just needed to rewire the $type re-assignment.

**Review by devinivy — APPROVED** [review]

**Pull request merged**